framework/CodeInterfaces/SERPENT/outputParser.py

```diff
@@ -0,0 +1,210 @@
+    @ Out, comptDict, dictionary, key=isotope
+                                  value=compDict
+  """
+  # check if percentCutoff value is valid
+  if percentCutoff < 0 or percentCutoff > 1:
+    raise ValueError('Percent has to be between 0 and 1')
+
+  # calculate atomicDensityCutoff
+  totalAtomicDensity = sum(compDict.values())
+  atomicDensityCutoff = percentCutoff * totalAtomicDensity
+
+  # delete list since cannot change dictionary during iteration
+  deleteList = []
+  for key, atomDensity in compDict.items():
+    if atomDensity < atomicDensityCutoff:
+      deleteList.append(key)
+
+  # delete the isotopes with less than percentCutoff
+  for isotope in deleteList:
+    del compDict[isotope]
+
+  return compDict
+
+
+
+def bumatRead(bumatFile, percentCutoff):
+  """
+  Reads serpent .bumat output file and stores the composition in a dictionary.
+    @ In, bumatFile, string, bumat file path
+    @ In, percentCutoff, float, cutoff threshold for ignoring isotopes (0 -1)
+    @ Out, compDict, dictionary, key=isotope
+                                value=atomic density
+  """
+  with open(bumatFile) as f:
+    compLines = f.readlines()[5:]
+
+  compDict = {}
+  header = compLines[0]
+  for i in range(1, len(compLines)):
+    parsed = parseLine(compLines[i])
+    # isotope as key, atomic density as value
+    compDict[parsed[0]] = parsed[1]
+
+  compDict = filterTrace(compDict, percentCutoff)
+  return compDict
+
+
+def searchKeff(resFile):
+  """
+  Searches and returns the mean keff value in the .res file.
+    @ In, resFile, string, path to .res file
+    @ Out, keffDict, dictionary, key = keff or sd
+                                 value = list of keff or sd
+  """
+  with open(resFile) as f:
+    lines = f.readlines()
+
+  keffList = []
+  sdList = []
+
+  for i in range(0, len(lines)):
+    if 'IMP_KEFF' in lines[i]:
+      keffList.append(keffLineParse(lines[i])[0])
+      sdList.append(keffLineParse(lines[i])[1])
+
+  keffDict = {}
+  keffDict['keff'] = keffList 
+  keffDict['sd'] = sdList
+  return keffDict
+
+
+def keffLineParse(keffLine):
+  """
+  Parses through the anaKeff line in .res file.
+    @ In, keffLine, string, string from .res file listing IMPKEFF
+    @ Out, keffTuple, tuple, (mean IMPKEFF, sd of IMPKEFF)
+  """
+  start = keffLine.find('=')
+  newKeffLine = keffLine[start:]
+  start = newKeffLine.find('[')
+  end = newKeffLine.find(']')
+  
+  # +3 and -1 is to get rid of leading and trailing whitespace
+  keffSd = newKeffLine[start + 3:end - 1]
+  (keff, sd) = keffSd.split(' ')
+  keffTuple = (keff, sd)
+  return keffTuple
+
+
+def csvRenderDict(csvFilename, dictionary, header):
+  """
+  Renders csv given the dictionary column 1 = key, column 2 = value.
+    @ In, csvFilename, string, path of csv file to be created
+    @ In, dictionary, dictionary, dictionary to be rendered into csv file
+    @ In, header, list, list of length 2 of header strings
+    @ Out, bool, bool, True if successful
```

Comment:
If the return value isn't a named variable, give it the name of the function (`csvRenderDict` in this case).